GitUI/CommandsDialogs/FormPush.cs

```diff
@@ -502,15 +502,15 @@ private bool HandlePushOnExit(ref bool isError, FormProcess form)
 
                 if (forcePush)
                 {
-                    if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force"))
+                    if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force") && !form.ProcessArguments.EndsWith(" -f"))
                     {
                         if (GitCommandHelpers.VersionInUse.SupportPushForceWithLease)
                         {
-                            form.ProcessArguments = form.ProcessArguments.Replace("push", "push --force-with-lease");
+                            form.ProcessArguments = form.ProcessArguments + " --force-with-lease";
```

Comment:
I don't think appending to the end of the args is the right thing to do. At this point, args will look something like:

```
push --recurse-submodules=check --progress "origin" refs/heads/test-force-push:refs/heads/test-force-push
```

We usually want to avoid adding args after we specify the remote and name of branch. I'm surprised this works.

I tried something locally, and this seems to work:

```csharp
                if (forcePush)
                {
                    if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force"))
                    {
                        string forceArg = GitCommandHelpers.VersionInUse.SupportPushForceWithLease? " --force-with-lease" : " -f";
                        form.ProcessArguments = form.ProcessArguments.Insert(form.ProcessArguments.IndexOf("push") + "push".Length, forceArg);
                    }

```

Or if we can assume "force" is always the first value in the args list, it can be simplified to:

```csharp
                if (forcePush)
                {
                    if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force"))
                    {
                        string forceArg = GitCommandHelpers.VersionInUse.SupportPushForceWithLease? " --force-with-lease" : " -f";
                        form.ProcessArguments = form.ProcessArguments.Insert("push".Length, forceArg);
                    }
```